File: frontend/property-pro-client/src/types/payment.ts
export interface Payment {
  paymentId: number
  leaseId: number
  tenantId: number
  tenantFirstName: string
  tenantLastName: string
  unitId: number
  unitNumber: string
  propertyName: string
  amountPaid: number
  paymentDate: string
  paymentMethod: string
  status: PaymentStatus
  periodMonth: number
  periodYear: number
  lateFee: number
  notes?: string
  receiptNumber: string
  createdAt: string
}

export type PaymentStatus = 'Paid' | 'Partial' | 'Late' | 'Pending'
export const PAYMENT_STATUSES: PaymentStatus[] = ['Paid', 'Partial', 'Late', 'Pending']

export type PaymentMethod = 'E-Transfer' | 'Cash' | 'Cheque' | 'Bank Transfer' | 'Credit Card' | 'Other'
export const PAYMENT_METHODS: PaymentMethod[] = ['E-Transfer', 'Cash', 'Cheque', 'Bank Transfer', 'Credit Card', 'Other']

export const MONTHS = [
  'January', 'February', 'March', 'April', 'May', 'June',
  'July', 'August', 'September', 'October', 'November', 'December',
]

File: backend/PropertyPro.API/Models/Payment.cs
namespace PropertyPro.API.Models;

public class Payment
{
    public int PaymentId { get; set; }
    public int LeaseId { get; set; }
    public Lease Lease { get; set; } = null!;
    public int LandlordId { get; set; }
    public User Landlord { get; set; } = null!;
    public decimal AmountPaid { get; set; }
    public DateTime PaymentDate { get; set; }
    public string PaymentMethod { get; set; } = "E-Transfer"; // E-Transfer | Cash | Cheque | Bank Transfer | Credit Card | Other
    public string Status { get; set; } = "Paid"; // Paid | Partial | Late | Pending
    public int PeriodMonth { get; set; }
    public int PeriodYear { get; set; }
    public decimal LateFee { get; set; }
    public string? Notes { get; set; }
    public string ReceiptNumber { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}
